File: frontend/receiptfly-web/src/utils/idGenerator.ts
import { v4 as uuidv4 } from 'uuid';

/**
 * レシートIDを生成
 * @returns receipt-{uuid} 形式のID
 */
export function generateReceiptId(): string {
  return `receipt-${uuidv4()}`;
}

/**
 * 取引明細項目IDを生成
 * @returns transaction-{uuid} 形式のID
 */
export function generateTransactionItemId(): string {
  return `transaction-${uuidv4()}`;
}

File: backend/Receiptfly.Application/Helpers/IdGenerator.cs
namespace Receiptfly.Application.Helpers;

/// <summary>
/// ID生成ヘルパー
/// </summary>
public static class IdGenerator
{
    /// <summary>
    /// レシートIDを生成
    /// </summary>
    /// <returns>receipt-{uuid} 形式のID</returns>
    public static string GenerateReceiptId()
    {
        return $"receipt-{Guid.NewGuid()}";
    }

    /// <summary>
    /// 取引明細項目IDを生成
    /// </summary>
    /// <returns>transaction-{uuid} 形式のID</returns>
    public static string GenerateTransactionItemId()
    {
        return $"transaction-{Guid.NewGuid()}";
    }
}

